javascript/code-challenges/challenges-06.test.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named getNames that, given an array of people objects, uses map to return an array of names reversed.
+
+For example:
+[
+{
+  name: 'lloyd',
+  age: 32,
+  shoeSize: 12
+},
+{
+  name: 'jamie',
+  age: 21,
+  shoeSize: 8
+}
+]
+
+Returns: ['dyoll', 'eimaj'];
+------------------------------------------------------------------------------------------------ */
+
+const getNames = (arr) => arr.map(str=>str.name.split('').reverse().join(''));
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function that appends ' The end.' to a string, and returns the modified string. The original source string should not be modified.
+
+------------------------------------------------------------------------------------------------ */
+
+const appendTheEnd = (str) => {
+  let newStr = `${str} The end.`;
+  return newStr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that accepts an array and copies the first element to the end of the array. The change should be reflected in the source array that was passed in to the function. That is, the function should modify the array 'in place'.
+
+Do not use a return statement.
+
+For example:
+const a = [1, 2, 3];
+appendFirstToLast(a);
+console.log(a) prints [1, 2, 3, 1]
+------------------------------------------------------------------------------------------------ */
+
+const appendFirstToLast = (arr) => {arr.push(arr[0]);};
```

Comment:
When doing a single line with an implicit return, you don't need the curly brackets.

```js
const appendFirstToLast = arr => arr.push(arr[0]);
```